x-pack/plugins/apm/public/components/shared/unified_search_bar/index.tsx

```diff
@@ -269,13 +275,20 @@ export function UnifiedSearchBar({
       };
 
       if (isUpdate) {
+        action = 'update';
         history.push({
           ...location,
           search: fromQuery(newSearchParams),
         });
       } else {
+        action = 'refresh';
```

Comment:
I am not 100% sure if that's a `refresh`.

You are initialising the action as `submit` but in this handler setting it to either `update` or `refresh`. This means `update` would never be passed as its either `update` or `refresh`

Reply:
In my opinion, there should be no `update` state as its always a `submit` or `refresh`

Reply:
> This means update would never be passed as its either update or refresh

Do you mean submit? I've noticed it while testing it

Reply:
Yup, it's either a `submit` (something new has been searched) or a `refresh` (existing page needs to be reloaded).